oscarlab/graphene
# Remove `libos.entrypoint` from PAL

## Description of the changes 

This is a follow-up to @mkow's earlier loader refactoring.

Unfortunately I had to remove the executable name from log prefix (`log_setprefix`), because the initialization happens too early to even parse it from manifest. I could restore it by initializing twice, or passing it from PAL, but unless you consider it important, I'd rather keep it that way - I don't see a non-hacky solution.

## How to test this PR? 

Apart from existing tests, you can try running something with `loader.log_level = "trace"` to see new log prefix.

## Next steps

As mentioned in #2415, I want to convert `libos.entrypoint` to be a Graphene path to executable, not host URI. But it's a breaking change, and requires rewriting all tests/examples, so I'll keep it for a separate PR.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/oscarlab/graphene/2416)
